lib/Metadata/BaseMetadataFactory.php

```diff
@@ -146,11 +156,30 @@ public function getMetadataForPhpcrNode(NodeInterface $node)
      */
     private function getMetadata($mapping)
     {
+        $mapping = array_merge(array(
```

Comment:
This needs to be cached locally. (and eventually in a persistent cache, but the overhead should be minimal here as we read directly from an array).